File: MegaHerdt.Models/Models/Provider.cs
using System.ComponentModel.DataAnnotations;

namespace MegaHerdt.Models.Models
{
    public class Provider
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        [Required]
        public string Phone { get; set; }
        [EmailAddress]
        public string Email { get; set; }
        public bool Enabled { get; set; } = true;
        public List<ArticleProvider> ArticlesProviders { get; set; }
    }
}

File: MegaHerdt.FrontEnd/MegaHerdt/src/app/models/Provider/Provider.ts
export class Provider{
    id: number;
    name: string;
    phone: string;
    email: string;
    constructor(id: number = 0, name: string = '', phone: string = '', email: string = ''){
        this.id = id;
        this.name = name;
        this.phone = phone;
        this.email = email;
    }
}